pandas/core/arrays/categorical.py

```diff
@@ -457,11 +460,14 @@ def _formatter(self, boxed=False):
         # Defer to CategoricalFormatter's formatter.
         return None
 
-    def copy(self):
+    def copy(self, deep: bool = False):
         """
         Copy constructor.
         """
-        return self._constructor(values=self._codes.copy(),
+        values = self._codes
+        if deep:
```

Comment:
this is the separate issue you reported yes?

Reply:
Yes, this is #26995.  This is not orthogonal though, needs to be fixed for this POC to work.